### ISSUE TITLE
TensorFlow builds fail on JURECA and JUWELS

### ISSUE THREAD

**Issue by SebastianAchilles** [easybuilders/easybuild-easyconfigs#13967]
Follow up on the failing test reports in https://github.com/easybuilders/easybuild-easyconfigs/pull/13877

Building TensorFlow 2.4.1, 2.5.0 and 2.6.0 fails on JURECA and JUWELS, but not on my minimalist rocky 8.4 container.

## EB Config and system info
- [JURECA: AMD EPYC 7742 + RTX 8000 or A100](https://gist.github.com/SebastianAchilles/5a534b6e3e3b6a6d53843dcb0c066bf9)
- [JUWELS: Intel(R) Xeon(R) Gold 6148 + P100 or V100](https://gist.github.com/SebastianAchilles/b724907d2f14f93b245581916be7605d)
- [rocky 8.4 Container: Intel(R) Core(TM) i7-6900K](https://gist.github.com/SebastianAchilles/1b589e1537934e3c91747cb2b1769fd2)

## Test Reports:

- [JURECA TensorFlow 2.4.1, 2.5.0, 2.6.0 foss](https://gist.github.com/30e1635a3df36dbacd54ee39fd849d33) **FAILED**
- [JURECA TensorFlow 2.4.1, 2.5.0, 2.6.0 foss + CUDA](https://gist.github.com/f2f4685df9f5ac3ae5cd0ad95607e000) **FAILED**
- [JUWELS TensorFlow 2.4.1, 2.5.0, 2.6.0 foss](https://gist.github.com/4912ce9c120dae86edaec033602d36f4) **FAILED**
- [JUWELS TensorFlow 2.4.1, 2.5.0, 2.6.0 foss + CUDA](https://gist.github.com/95ea43ca8f4f92d405dca60a849fcea2) **FAILED**
- [rocky 8.4 Container TensorFlow 2.4.1, 2.5.0, 2.6.0 foss](https://gist.github.com/60bea5a5f88bb16689661572af730392) **SUCCESS**

I am trying to understand why the builds fail on JURECA and JUWELS. And whether this is caused by to the specific configuration of these systems. Maybe @Flamefire or @boegel can give me a hint to better understand the problem?

**Comment by boegel** [comment]
@SebastianAchilles Anything useful in the logs for the failing `csv_dataset_test`?

**Comment by SebastianAchilles** [comment]
```
2021-09-09 20:56:46.792936: F tensorflow/core/platform/default/env.cc:73] Check failed: ret == 0 (11 vs. 0)Thread tf_data_private_threadpool creation via pthread_create() failed.
Fatal Python error: Aborted
```
This looks like there is a problem with the creation of the threads. I assume this is caused by our (too?) strict limitations on the login nodes.
On the backend nodes I can not build TensorFlow at the moment because, `UnZip` not in our lightweight OS image, but is required for building. I have already opened PRs for this:
- https://github.com/easybuilders/easybuild-easyconfigs/pull/13958
- https://github.com/easybuilders/easybuild-easyconfigs/pull/13959
- https://github.com/easybuilders/easybuild-easyconfigs/pull/13960

**Comment by Flamefire** [comment]
> With UnZip added, I get a bit further on the backend node

What was the error when that was not added?

> But it seems like openssl-devel is still required as OS dependency.

Bazel sends its greetings... Will send an easyblock PR to fix this.

**Comment by branfosj** [comment]
> With UnZip added, I get a bit further on the backend node

Did it fail on Dill? We use `'source_tmpl': '%(name)s-%(version)s.zip',` for that - so the UnZip would be needed to unpack that if you do not have that installed in the OS. This works because the build dependencies are loaded before extensions are unpacked.

Also, does SciPy-bundle fail in the same way? There numpy is a `SOURCE_ZIP`.

**Comment by SebastianAchilles** [comment]
> > With UnZip added, I get a bit further on the backend node
> 
> Did it fail on Dill?

Yes, it failed on Dill. On the lightweight OS of the backend nodes we do not have `UnZip` installed.

> Also, does SciPy-bundle fail in the same way? There numpy is a `SOURCE_ZIP`.

Yes, I got the same error `unzip not found` with `SciPy-bundle` on our backend nodes.


> > But it seems like openssl-devel is still required as OS dependency.
> 
> Bazel sends its greetings... Will send an easyblock PR to fix this.

Thank you!

**Comment by Flamefire** [comment]
@SebastianAchilles Can you test with https://github.com/easybuilders/easybuild-easyblocks/pull/2575?

**Comment by SebastianAchilles** [comment]
> @SebastianAchilles Can you test with [easybuilders/easybuild-easyblocks#2575](https://github.com/easybuilders/easybuild-easyblocks/pull/2575)?

I will start a test report after 6PM. Today we have maintenance day on all systems.
